### PR TITLE
fix(react-storybook-addon-export-to-sandbox): ignore relative paths for creating story dependencies and remove relative imports within babel plugin if they dont point to barrel file

### LINKED ISSUE #30668: [Bug]: Open in Stackblitz fails for some stories
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
-
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://stackblitz.com/edit/ktkhdu?file=%2Fhome%2F.npm%2F_logs%2F2024-02-29T22_31_20_567Z-debug-0.log

### Bug Description

Open [DateTime -> Freeform custom parsing](https://react.fluentui.dev/?path=/docs/compat-components-timepicker--default#freeform-custom-parsing) story in Stackblitz

## Actual Behavior
Stackblitz fails to install dependencies:
```sh
~/projects/wepu2l--run
❯ npm install && npm run dev
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "." of package ".@latest": name cannot start with a period.

npm ERR! A complete log of this run can be found in: /home/.npm/_logs/2024-02-29T22_30_24_230Z-debug-0.log
```

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by Hotell**
### addon - relative import paths are ignored when building package.json dependencies

previously relative import `import story from "./TimePickerFreeformCustomParsing.md";` created invalid package definition `".": "latest",`.

This will be a noop now ( relative imports are ignored )

### babel-plugin - resolution of relative import paths within babel plugin has been changed

While we don't leverage this behaviour anymore, outside of core fluent it might be still used thus the functionality is still there but the logic was changed to explicitly resolve package.json only when we run into imports from barrel file (`./index`, `../../../index` ) etc.

If there is any other relative import present, we remove it from output - this might create invalid code example if symbols from that import are being used in story rendering logic. 

ATM we will only `console.warn` developer during inner loop  ( up to discussion if we should throw error and break storybook build ), because there might be valid cases that might need to import from `.md` file if the content story description is more verbose (like seen within https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-timepicker-compat/stories/TimePicker/TimePickerFreeformWithErrorHandling.md ). 

image

## Related Issue(s)

- Fixes #30668

### CODE REVIEW

File: packages/react-components/babel-preset-storybook-full-source/src/modifyImports.ts [hunk 1]
```diff
@@ -49,7 +49,24 @@ export function modifyImportsPlugin(babel: typeof Babel, options: BabelPluginOpt
       // eslint-disable-next-line @typescript-eslint/naming-convention
       ImportDeclaration(path, pluginState) {
         let importSource = path.node.source;
-        if (importSource.value.startsWith('.')) {
```
Comment (Hotell) [inline packages/react-components/babel-preset-storybook-full-source/src/modifyImports.ts:52]: this was naive logic, as any import paths was considered to be import from package API which might not be the case.

File: packages/react-components/babel-preset-storybook-full-source/src/modifyImports.test.ts [hunk 2]
```diff
@@ -5,14 +5,22 @@ import pluginTester from 'babel-plugin-tester';
...
-  pluginOptions: {
-    '@fluentui/react-button': defaultDependencyReplace,
-    '@fluentui/react-menu': defaultDependencyReplace,
-    '@fluentui/react-link': defaultDependencyReplace,
-    '@fluentui/react-unstable-component': { replace: '@fluentui/react-components/unstable' },
-  },
-  pluginName: PLUGIN_NAME,
-  plugin,
+describe(PLUGIN_NAME, () => {
+  beforeEach(() => {
```
Comment (Hotell) [inline packages/react-components/babel-preset-storybook-full-source/src/modifyImports.test.ts:9]: tried `setup` but didn't work at all https://github.com/babel-utils/babel-plugin-tester/tree/v10.1.0?tab=readme-ov-file#setup

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/0b3daaa393add4d5fc559b834b03145fa2853c79'>0b3daaa393add4d5fc559b834b03145fa2853c79</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by Hotell** [comment]
our options regarding babel plugin and non barel relative imports :

1. leave it as is - which works within stackblitz ATM
 - PROS:
   - none 
 - CONS:
   - output provided is invalid javascript module that might break in the future ( based on bundling used )
3. apply changes within this PR - remove relative imports invalid for Example story and warn developer about it
  - PROS: 
    - allows extracting more verbose description in separate `.md` 
    - allow inconsistent story authoring for our documentation
  - CONS: 
    - potential hazard of shipping invalid `Examples` to stackblitz 
4. throw build error if present 
  - PROS:
    - enforces consistent story authoring
    - makes sure that our `Examples` shipped to stackblitz always work
  - CONS: 
    - not the best DX when verbose docs per story are needed. Needs to be authored as String array line by line
    - image

**Review by tudorpopams — APPROVED** [review]

**Review by TristanWatanabe — APPROVED** [review]
Seems like option 2 which is implemented in this PR resolves the stackblitz related issue and I think that this solution is a good workaround. Since the issue is isolated to stackblitz exports, I don't think it's necessary to enforce a new storybook doc authoring process as option 3 suggests. Probably something we can revisit if more cases like this pop up in the future

**Pull request merged**